src/helpers/pollUpdates.ts

```diff
@@ -32,8 +32,8 @@ const processEnvironment = (store: Store, environment: Environment) => {
   const releaseId = environment.release_id;
 
   if (
-    environment.release_id !== RELEASE_ID
-    && (trustRelease() || previousObservedReleaseId !== releaseId)
+    (trustRelease() && environment.release_id !== RELEASE_ID)
+    || (previousObservedReleaseId && previousObservedReleaseId !== releaseId)
   ) {
```

Comment:
You've assigning `environment.release_id` to `releaseId` on line 32, but only using once in the condition and using the original `environment.release_id` in the other spot.

Also seems like you wouldn't need to test if `previousObservedReleaseId` is defined.  Even if it's null/undefined it wouldn't match `releaseId`, right? 

Seems like you could:
```js
 (trustRelease() && releaseId !== RELEASE_ID) || (releaseId  !== previousObservedReleaseId)
```

Reply:
if i didn't null check the `previousObservedReleaseId` then on the initial pass `null` would be different from `environment.release_id` and it'd show the toast, which is what i'm trying to avoid. with the null check it noops on the first pass and by the second pass they'll equal one another (or it'll have changed, in which case showing the toast is correct)

Reply:
oh yep that makes sense. thanks!